stacks/shared-ecs/asg.yml

```diff
@@ -99,6 +99,333 @@ Resources:
+            logger.debug("SNS ARN %s", snsArn)
+
+            # Get list of container instance IDs from the clusterName
+            clusterListResp = ecsClient.list_container_instances(cluster=clusterName)
+            logger.debug("list container instances response %s", clusterListResp)
+
+            # If the event received is instance terminating...
+            if "LifecycleTransition" in message.keys():
+                logger.debug("message autoscaling %s", message["LifecycleTransition"])
+                if (
+                    message["LifecycleTransition"].find("autoscaling:EC2_INSTANCE_TERMINATING")
+                    > -1
+                ):
+
+                    # Get lifecycle hook name
+                    lifecycleHookName = message["LifecycleHookName"]
+                    logger.debug("Setting lifecycle hook name %s ", lifecycleHookName)
+
+                    # Check if there are any tasks running on the instance
+                    tasksRunning, tmpMsgAppend = checkContainerInstanceTaskStatus(Ec2InstanceId)
+                    logger.debug("Returned values received: %s ", tasksRunning)
+                    if tmpMsgAppend is not None:
+                        message.update(tmpMsgAppend)
+
+                    # If tasks are still running...
+                    if tasksRunning == 1:
+                        logger.info("SQS delayed callback to %s...", TopicArn)
+                        time.sleep(5)
+                        msgResponse = publishToSNS(message, TopicArn)
+                        logger.debug(
+                            "msgResponse %s and time is %s", msgResponse, datetime.datetime
+                        )
+                    # If tasks are NOT running...
+                    elif tasksRunning == 0:
+                        # completeHook = 1
+                        logger.debug(
+                            (
+                                "Setting lifecycle to complete;No tasks are running "
+                                "on instance, completing lifecycle action...."
+                            )
+                        )
+
+                        try:
+                            response = asgClient.complete_lifecycle_action(
+                                LifecycleHookName=lifecycleHookName,
+                                AutoScalingGroupName=asgGroupName,
+                                LifecycleActionResult="CONTINUE",
+                                InstanceId=Ec2InstanceId,
+                            )
+                            logger.info(
+                                "Response received from complete_lifecycle_action %s", response
+                            )
+                            logger.info("Completedlifecycle hook action")
+                        except Exception as e:
+                            print(str(e))
+      MemorySize: 256
+      Policies:
+        - Statement:
+            - Action:
+                - ec2:DescribeInstances
```

Comment:
Don't know if it matters, but I dropped the [logging permissions](https://github.com/PRX/Infrastructure/blob/bcdbefed9ad2cddf5bc8d1112b254918789e06c1/stacks/ecs.yml#L695) here.